src/unix/async.c

```diff
@@ -58,7 +58,11 @@ int uv_async_init(uv_loop_t* loop, uv_async_t* handle, uv_async_cb async_cb) {
 
 
 int uv_async_send(uv_async_t* handle) {
-  if (uv__async_make_pending(&handle->pending) == 0)
+  /* Do a cheap read first. */
+  if (ACCESS_ONCE(int, handle->pending) != 0)
```

Comment:
For posterity, I'm not aware of any architectures where aligned loads and stores to the same memory address are not atomic.  It's only with multiple addresses that you have to start worrying about the architecture's memory model.